qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -158,29 +164,68 @@ def __init__(
 
         self._quantum_instance = None
         if quantum_instance is not None:
+            warnings.warn(
+                "The quantum_instance argument has been superseded by the sampler argument. "
+                "This argument will be deprecated in a future release and subsequently "
+                "removed after that.",
+                category=PendingDeprecationWarning,
+                stacklevel=2,
+            )
             self.quantum_instance = quantum_instance
```

Comment:
Does this raise two deprecation warnings? One right here and one in the quantum_instance setter? If yes, you could check the other algorithms which use `warnings.filterwarnings` to suppress the second deprecation warning 🙂